ulog/builder_test.go

```diff
@@ -147,6 +148,49 @@ func TestConfiguredLoggerWithSentryError(t *testing.T) {
 	testSentry(t, "invalid_dsn", false)
 }
 
+func TestMetricsHook(t *testing.T) {
+	// TODO(glib): logging init needs to be fixed
+	// Any tests that run in development environment can't configure the logger,
+	// it just creates the default one...
+	defer env.Override(t, config.EnvironmentKey(), "wat?")()
+
+	s, r := metrics.NewTestScope()
+	l := Builder().WithScope(s).Build()
+
+	r.CountersWG.Add(1)
+	l.Warn("Warning log!")
+	r.CountersWG.Wait()
+
+	assert.Equal(t, 1, len(r.Counters))
+}
+
+func TestLoggingDisabled(t *testing.T) {
+	tests := []struct {
+		res  bool
+		data []byte
+	}{
+		{
+			false,
+			[]byte("logging:\n  lalala: foo"),
+		},
+		{
+			false,
+			[]byte("logging:\n  disableMetrics: false"),
+		},
+		{
+			true,
+			[]byte("logging:\n  disableMetrics: true"),
+		},
+	}
+
+	for _, tt := range tests {
+		logCfg := Configuration{}
+		logYamlCfg := config.NewYAMLProviderFromBytes(tt.data)
+		logYamlCfg.Get("logging").PopulateStruct(&logCfg)
```

Comment:
i'll remove the test because you're right it doesn't help. But testing behavior right now is not easy - since zap doesn't let you peek into the hooks or options that were used to create it. I'll separate the `Build()` method into two parts, so we can assert which options zap is about to be created with.